### ISSUE TITLE
Check how these perform with server side rendering

### ISSUE THREAD

**Issue by prateekbh** [prateekbh/preact-material-components#6]
Most of initialization code is written in componentDidMount, verify how the noJS version of these things work.
or even how does the transition from server rendered to client rendering looks like

**Comment by cromefire** [comment]
I will look at this shortly

**Comment by prateekbh** [comment]
yasss thank you for all the love ❤️

**Comment by cromefire** [comment]
Not doing all maybe in a few days, on some you need to decide

**Comment by cromefire** [comment]
```
Unhandled promise rejection ReferenceError: MutationObserver is not defined
    at Object.registerValidationAttributeChangeHandler (eval at <anonymous> (/path/to/project/node_modules/prerender-loader/src/index.js:192:21), <anonymous>:7074:26)
    at MDCTextFieldFoundation.init (eval at <anonymous> (/path/to/project/node_modules/prerender-loader/src/index.js:192:21), <anonymous>:5442:48)
    at MDCTextField.MDCComponent (eval at <anonymous> (/path/to/project/node_modules/prerender-loader/src/index.js:192:21), <anonymous>:379:22)
    at new MDCTextField (eval at <anonymous> (/path/to/project/node_modules/prerender-loader/src/index.js:192:21), <anonymous>:6884:249)
    at TextFieldInput.eval (eval at <anonymous> (/path/to/project/node_modules/prerender-loader/src/index.js:192:21), <anonymous>:13048:42)
    at renderComponent (eval at <anonymous> (/path/to/project/node_modules/prerender-loader/src/index.js:192:21), <anonymous>:15345:37)
    at setComponentProps (eval at <anonymous> (/path/to/project/node_modules/prerender-loader/src/index.js:192:21), <anonymous>:15204:4)
    at buildComponentFromVNode (eval at <anonymous> (/path/to/project/node_modules/prerender-loader/src/index.js:192:21), <anonymous>:15370:3)
    at idiff (eval at <anonymous> (/path/to/project/node_modules/prerender-loader/src/index.js:192:21), <anonymous>:14925:10)
    at innerDiffNode (eval at <anonymous> (/path/to/project/node_modules/prerender-loader/src/index.js:192:21), <anonymous>:15044:12)
```
_(prerender-loader)_

Challenge accepted

**Comment by cromefire** [comment]
Ref #929

**Comment by cromefire** [comment]
Fixed by #982
